shared/src/commonMain/kotlin/io/customer/shared/tracking/queue/BackgroundQueue.kt

```diff
@@ -0,0 +1,204 @@
+package io.customer.shared.tracking.queue
+
+import io.customer.shared.Platform
+import io.customer.shared.common.CustomAttributes
+import io.customer.shared.database.TrackingTaskQueryHelper
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.tracking.api.model.Device
+import io.customer.shared.tracking.constant.MetricEvent
+import io.customer.shared.tracking.constant.TrackingType
+import io.customer.shared.tracking.model.Activity
+import io.customer.shared.tracking.model.Task
+import io.customer.shared.util.DateTimeUtil
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobDispatcher
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.runOnBackground
+import io.customer.shared.work.runOnMain
+import kotlinx.coroutines.Job
+
+/**
+ * Background queue is responsible for queuing tasks to trigger when needed. It works asynchronously
+ * and does not blocks the caller for any operation. Caller should provide [TaskResultListener] to
+ * listen on results for any request.
+ */
+interface BackgroundQueue {
+    fun queueIdentifyProfile(
+        profileIdentifier: String,
+        attributes: CustomAttributes,
+        listener: TaskResultListener<Unit>? = null,
+    ): Job
+
+    fun queueTrack(
+        profileIdentifier: String?,
```

Comment:
Why is profile identifier optional?

Reply:
With KMM, we can keep queueing events until profile is identified. As soon as the profile is identified, we [update all anonymous events with current profile](https://github.com/customerio/mobile-shared/blob/945c9cb4ca5e02374ef5f3a305b254dbc337a58a/shared/src/commonMain/kotlin/io/customer/shared/database/TrackingTaskQueryHelper.kt#L98-L106). Since this is a behavior change, it can be put behind configuration and controlled by the SDKs. But KMM does not restrict to identify profile before adding events. It will keep them locally, and [send it only after they have been tagged with some profile](https://github.com/customerio/mobile-shared/blob/945c9cb4ca5e02374ef5f3a305b254dbc337a58a/shared/src/commonMain/sqldelight/io/customer/shared/database/TrackingTask.sq#L55-L59).